rust/datafusion/src/datasource/parquet.rs

```diff
@@ -62,17 +64,37 @@ impl TableProvider for ParquetTable {
         self.schema.clone()
     }
 
+    fn supports_filter_pushdown(
+        &self,
+        _filter: &Expr,
+    ) -> Result<TableProviderFilterPushDown> {
+        Ok(TableProviderFilterPushDown::Inexact)
+    }
+
     /// Scan the file(s), using the provided projection, and return one BatchIterator per
     /// partition.
     fn scan(
         &self,
         projection: &Option<Vec<usize>>,
         batch_size: usize,
-        _filters: &[Expr],
+        filters: &[Expr],
     ) -> Result<Arc<dyn ExecutionPlan>> {
+        let predicate = if filters.is_empty() {
+            None
+        } else {
+            Some(
+                filters
```

Comment:
Could you add a comment explaining the logic here? It isn't immediately obvious to me.

Reply:
Immediately after posting that comment I see how it works now, but I think a comment would still be helpful

Reply:
👍